Code/Framework/AzCore/AzCore/Debug/Profiler.inl

```diff
@@ -20,50 +20,61 @@ namespace AZ::Debug
 
     template<typename... T>
     void ProfileScope::BeginRegion(
-        [[maybe_unused]] Budget* budget, [[maybe_unused]] const char* eventName, [[maybe_unused]] T const&... args)
+        [[maybe_unused]] Budget* budget, [[maybe_unused]] bool budgetTotal, [[maybe_unused]] const char* eventName, [[maybe_unused]] T const&... args)
     {
     #if !defined(_RELEASE)
         if (budget)
         {
-            Platform::BeginProfileRegion(budget, eventName, args...);
-
-            budget->BeginProfileRegion();
+            if (!budgetTotal)
+            {
+                Platform::BeginProfileRegion(budget, eventName, args...);
 
-            if (auto profiler = AZ::Interface<Profiler>::Get(); profiler)
+                if (auto profiler = AZ::Interface<Profiler>::Get(); profiler)
+                {
+                    profiler->BeginRegion(budget, eventName, sizeof...(T), args...);
+                }
+            }
+            else
             {
-                profiler->BeginRegion(budget, eventName, sizeof...(T), args...);
+                budget->BeginProfileRegion();
             }
```

Comment:
This could be simplified if we used a separate class(instead of ProfileScope) as mentioned in the comment in Profiler.h above.